packages/p2p/src/components/orders/order-details/order-details.jsx

```diff
@@ -49,122 +56,144 @@ const OrderDetails = ({ order_information, chat_info }) => {
     };
 
     React.useEffect(() => {
-        if (isMounted()) {
-            if (!channel_url) {
-                // If order_information doesn't have channel_url this is a new order
-                // and we need to instruct BE to create a chat on Sendbird's side.
-                requestWS({ p2p_chat_create: 1, order_id: id }).then(response => {
-                    if (response.error) {
-                        // TODO: Handle error.
-                        return;
-                    }
+        const disposeListeners = sendbird_store.registerEventListeners();
```

Comment:
Since we're using plenty of `sendbird_store .` snd we're already destructuring on line 48, can we also destructure other functions and observables?

Reply:
For basic data types (such as strings and numbers) and bound actions (`action.bound` decorated functions) we can destructure.

- For basic data types because we don't reference `this`.
- For bound actions because MobX ensures the `this` context is correct.

If you destructure a class method from a specific class, it'll lose its original context.

Once we write a new connector function we'll get rid of this bloat.

Reply:
Alrighty 😄